### ISSUE TITLE
Security: Don't eval data from socket if writable by others

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/next#671]
And should we add an option to enable/disable evaluation via socket?

**Comment by jmercouris** [comment]
Should a malicious actor already have write access on your file system, I think you are already compromised. They could for example just replace Next with their own version... that said, it’s just a flag, so why not :-)

**Comment by Ambrevar** [comment]
There is a more trivial case: Consider multi-user systems where Alice
and Bob are part of the "users" group.  If the socket is group writable,
then Alice can compromise the Next that Bob is running (e.g. over SSH).

**Comment by jmercouris** [comment]
How could we determine if a socket is writable by others? I think our only solution is most likely to have the toggle to enable/disable evaluation via socket

**Comment by Ambrevar** [comment]
With osicat:

```
> (osicat:file-permissions "path/to/file")
(:USER-READ :USER-WRITE :GROUP-READ :OTHER-READ)
```

**Comment by jmercouris** [comment]
Yes, but how do we know about which other users can write, we would need to look at their groups and permissions for every user?

**Comment by Ambrevar** [comment]
We don't need to loop over users, we only need to check if only
:USER-READ and :USER-WRITE are in the file permissions.  Does that make sense?

**Comment by jmercouris** [comment]
That does not make sense to me. I think :user-write would always be true, otherwise, how could we write to the socket?

**Comment by Ambrevar** [comment]
Let me rephrase:

we allow remote execution if only :USER-WRITE and :USER-READ or in the
permissions.
In particular, we disallow execution if :OTHER-WRITE and :GROUP-WRITE is
in the permissions.

**Comment by jmercouris** [comment]
Ah! OK, I understand, I can do this

**Comment by Ambrevar** [comment]
Fixed with efe8e8344797351736968ebc811a92c9a3cfc336.
